src/mlpack/core/tree/cover_tree/cover_tree_impl.hpp

```diff
@@ -547,6 +548,62 @@ CoverTree<MetricType, StatisticType, MatType, RootPointPolicy>::CoverTree(
   }
 }
 
+// Copy Assignment.
+template<
+    typename MetricType,
+    typename StatisticType,
+    typename MatType,
+    typename RootPointPolicy
+>
+CoverTree<MetricType, StatisticType, MatType, RootPointPolicy>&
+CoverTree<MetricType, StatisticType, MatType, RootPointPolicy>::
+operator=(const CoverTree& other)
+{
```

Comment:
We should add a stanza here and for the move constructor:

```
if (this == &other)
  return *this;
```

Copy and move operators can cause problems if we are doing them to the same object. :)